### PR TITLE
HIVE-28153: Flaky test TestConflictingDataFiles.testMultiFiltersUpdate

### PR DESCRIPTION
**Pull request by simhadri-g**
### What changes were proposed in this pull request?

Run the test with vectorization true

### Why are the changes needed?

IOContext gets shared in unit test causing non-vectorized to be flaky. 

### Is the change a dependency upgrade?

No

### How was this patch tested?

Unit test TestConflictingDataFiles

### CODE REVIEW

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestConflictingDataFiles.java [hunk 1]
```diff
@@ -44,6 +45,7 @@ public class TestConflictingDataFiles extends HiveIcebergStorageHandlerWithEngin
 
   @Before
   public void setUpTables() {
+    Assume.assumeTrue(isVectorized && testTableType == TestTables.TestTableType.HIVE_CATALOG);
```
Comment (deniskuzZ) [inline iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestConflictingDataFiles.java:48]: use this instead
````
@Override
protected void validateTestParams() {
  Assume.assumeTrue(fileFormat == FileFormat.PARQUET && isVectorized &&
      testTableType == TestTables.TestTableType.HIVE_CATALOG);
}
````

Comment (simhadri-g) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestConflictingDataFiles.java]: Fixed

**Comment by zhangbutao** [comment]
Should we do a flaky test to verify if the fix is good? http://ci.hive.apache.org/job/hive-flaky-check/

**Comment by simhadri-g** [comment]
@zhangbutao 
I have run the flaky test jenkins job on my fork and the test passed. 

http://ci.hive.apache.org/job/hive-flaky-check/835/console

**Comment by simhadri-g** [comment]
Hi @zhangbutao ,

I have started the flaky test jenkins job here: http://ci.hive.apache.org/job/hive-flaky-check/836/

**Comment by simhadri-g** [comment]
The flaky test check has passed: http://ci.hive.apache.org/job/hive-flaky-check/836/

**Review by deniskuzZ — APPROVED** [review]
+1

**Comment by simhadri-g** [comment]
Thanks @deniskuzZ and @zhangbutao  for the review :)

**Pull request merged**